src/main/java/hudson/plugins/promoted_builds/JobPropertyImpl.java

```diff
@@ -201,7 +216,11 @@ private void buildActiveProcess() throws IOException {
             // ensure that the name casing matches what's given in the activeProcessName
             // this is because in case insensitive file system, we may end up resolving
             // to a directory name that differs only in their case.
-            p.renameTo(getActiveProcessName(p.getName()));
+            String processName = p.getName();
+            String activeProcessName = getActiveProcessName(processName);
+            if (!activeProcessName.equals(processName)){
+            	p.renameTo(activeProcessName);
+            }
```

Comment:
It is actually related to the issue. Unfortunately p.renameTo results in an attempt to lock the promotion process and ultimately a deadlock.